trinity/buffer/reader/file_reader.py

```diff
@@ -2,55 +2,65 @@
 
 from typing import List, Optional
 
+import datasets
 import transformers
 from datasets import load_dataset
 
 from trinity.buffer.buffer_reader import BufferReader
-from trinity.common.config import BufferConfig, DatasetConfig
-from trinity.common.constants import (
-    AlgorithmType,
-    PromptType,
-    ReadStrategy,
-    StorageType,
-)
+from trinity.common.config import BufferConfig, StorageConfig
+from trinity.common.constants import AlgorithmType, PromptType, ReadStrategy, TaskType
 from trinity.common.experience import Experience
+from trinity.common.rewards import REWARD_FUNCTIONS
+from trinity.common.task import Task
+from trinity.common.workflows import WORKFLOWS
 
 
-class FileReader(BufferReader):
-    """Reader of the File buffer."""
+class FileReaderManager:
+    subclasses: dict = {}
 
-    def __init__(self, meta: DatasetConfig, config: BufferConfig) -> None:
-        assert meta.storage_type == StorageType.FILE
-        if meta.algorithm_type == AlgorithmType.SFT:
-            self.reader = SFTDataReader(meta, config)
-        elif meta.algorithm_type == AlgorithmType.DPO:
-            self.reader = DPODataReader(meta, config)
-        else:
-            # TODO: support read rollout task
-            raise ValueError(f"Unsupported algorithm type: {meta.algorithm_type}")
+    @classmethod
+    def register_subclass(cls, algorithm_type: AlgorithmType):
```

Comment:
Use  `Registry` in `trinity/regsitry.py` instead.

Reply:
ok, fixed